extra_data/keydata.py

```diff
@@ -366,6 +372,26 @@ def as_single_value(self, rtol=1e-5, atol=0.0, reduce_by='median'):
 
         return value
 
+    def as_unique_value(self):
+        """Retrieve a single value if unique.
+
+        Returns the single unique of this key for all trains provided
+        there is a single one.
+        """
+
+        data = self.ndarray()
+
+        if len(data) == 0:
+            raise NoDataError(self.source, self.key)
+
+        axis = 0 if np.issubdtype(self.dtype, np.number) else None
+        unique_values = np.unique(data, axis=axis)
```

Comment:
I'm OK with not supporting multidimensional object data for now, but I think we should throw an error (`NotImplementedError`?) rather than just doing something different with it.